features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/featureflags/internal/model/DatadogEvaluationContextTest.kt

```diff
@@ -0,0 +1,316 @@
+
+        // Then
+        assertThat(datadogContext).isNotNull
+        assertThat(datadogContext?.targetingKey).isEqualTo(targetingKey)
+        assertThat(datadogContext?.attributes).isEqualTo(
+            mapOf(
+                "stringAttr" to stringValue,
+                "numberAttr" to numberValue.toString(),
+                "booleanAttr" to booleanValue.toString()
+            )
+        )
+        verifyNoInteractions(mockInternalLogger)
+    }
+
+    @Test
+    fun `M filter unsupported types and log warnings W from() { mixed valid and invalid attributes }`(forge: Forge) {
+        // Given
+        val targetingKey = forge.anAlphabeticalString()
+        val validStringValue = forge.anAlphabeticalString()
+        val validNumberValue = forge.anInt()
+        val invalidListValue = listOf(1, 2, 3)
+        val invalidMapValue = mapOf("nested" to "not supported")
+
+        val publicContext = EvaluationContext(
+            targetingKey = targetingKey,
+            attributes = mapOf(
+                "validString" to validStringValue,
+                "validNumber" to validNumberValue,
+                "invalidList" to invalidListValue,
+                "invalidMap" to invalidMapValue
+            )
+        )
+
+        // When
+        val datadogContext = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        // Only valid attributes should be present
+        assertThat(datadogContext?.attributes).isEqualTo(
+            mapOf(
+                "validString" to validStringValue,
+                "validNumber" to validNumberValue.toString()
+            )
+        )
+
+        // Warnings should be logged for invalid types
+        verify(mockInternalLogger, times(2)).log(
+            eq(InternalLogger.Level.WARN),
+            eq(InternalLogger.Target.USER),
+            any<() -> String>(),
+            anyOrNull<Throwable>(),
+            any<Boolean>(),
+            anyOrNull<Map<String, Any?>>()
+        )
+    }
+
+    @Test
+    fun `M filter null values and log warnings W from() { attributes with null values }`(forge: Forge) {
+        // Given
+        val targetingKey = forge.anAlphabeticalString()
+        val validValue = forge.anAlphabeticalString()
+
+        val publicContext = EvaluationContext(
+            targetingKey = targetingKey,
+            attributes = mapOf(
+                "validAttr" to validValue,
+                "nullAttr" to null
+            )
+        )
+
+        // When
+        val datadogContext = DatadogEvaluationContext.from(publicContext, mockInternalLogger)
+
+        // Then
+        // Only non-null attributes should be present
+        assertThat(datadogContext?.attributes).isEqualTo(
+            mapOf(
+                "validAttr" to validValue
+            )
+        )
+
+        // Warning should be logged for null value
+        verify(mockInternalLogger, times(1)).log(
```

Comment:
`times(1)` is a default validation mode.

```suggestion
        verify(mockInternalLogger).log(
```